src/UKHO.ERPFacade.Common/PermitDecryption/BFFactory.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Runtime.Caching;
+
+namespace UKHO.ERPFacade.Common.PermitDecryption
+{
+    public interface IBFFactory
+    {
+        /// <summary>
+        /// Gets the specified key.
+        /// </summary>
+        /// <param name="key">The key.</param>
+        /// <param name="cachingOptions"></param>
+        /// <returns></returns>
+        IBlowfishAlgorithm Get(byte[] key, CachingOptions cachingOptions = CachingOptions.Cache);
+    }
+
+    [ExcludeFromCodeCoverage]
+    public class BFFactory : IBFFactory
```

Comment:
Please rename to BlowfishFactory, IBlowfishFactory etc

Reply:
Renamed